src/components/wiki.ts

```diff
@@ -25,101 +25,196 @@ async function* walk_dir(dir: string): AsyncGenerator<string> { // todo: duplica
 
 type WikiArticle = {
     title: string;
-    body: string | null;
-    fields: {name: string, value: string, inline: boolean}[],
+    body?: string;
+    fields: WikiField[],
     footer?: string;
-    set_author?: true;
-    no_embed?: true;
+    image?: string;
+    set_author: boolean;
+    no_embed: boolean;
 };
 
-export function parse_article(name: string | null, content: string): [WikiArticle, Map<string, string>] {
-    const data: Partial<WikiArticle> = {};
-    data.body = "";
-    data.fields = [];
-    const lines = content.split(/\r?\n/);
-    enum state { body, field, footer }
-    let code = false;
-    let current_state = state.body;
-    const article_aliases: Map<string, string> = new Map();
-    for(const [ i, line ] of lines.entries()) {
-        if(line.trim().startsWith("```")) {
-            code = !code;
+type WikiField = {
+    name: string;
+    value: string;
+    inline: boolean;
+}
```

Comment:
There are a couple places where semicolons are missing. I updated the eslint rules to catch these in the future.